build.rs

```diff
@@ -0,0 +1,159 @@
+use std::collections::BTreeMap;
```

Comment:
1. We should have comments on what this file does :)
2. I am not a super big fan of using Rust to build files :( We should use something like Bash or Python I think, as it's easier to read & contribute to?